modules/xfem/src/efa/EFAEdge.C

```diff
@@ -0,0 +1,419 @@
+//
+//  int this_node1_unique_index = ((int) edge_node1->category + 1) * edge_node1->id;
+//  int this_node2_unique_index = ((int) edge_node2->category + 1) * edge_node2->id;
+//  int other_node1_unique_index = ((int) other.edge_node1->category + 1) * edge_node1->id;
+//  int other_node2_unique_index = ((int) other.edge_node2->category + 1) * edge_node2->id;
+//  int this_min_index = std::min(this_node1_unique_index, this_node2_unique_index);
+//  int other_min_index = std::min(other_node1_unique_index, other_node2_unique_index);
+//
+//  if (this_min_index < other_min_index)
+//    return true;
+//  else if (this_min_index == other_min_index)
+//  {
+//    int this_max_index = std::max(this_node1_unique_index, this_node2_unique_index);
+//    int other_max_index = std::max(other_node1_unique_index, other_node2_unique_index);
+//
+//    if (this_max_index < other_max_index)
+//      return true;
+//  }
+//  return false;
+//}
+
+void
+EFAEdge::addIntersection(double position, EFANode * embedded_node_tmp, EFANode * from_node)
+{
+  _embedded_nodes.push_back(embedded_node_tmp);
+  if (from_node == _edge_node1)
+    _intersection_x.push_back(position);
+  else if (from_node == _edge_node2)
+    _intersection_x.push_back(1.0 - position);
+  else
+    EFAError("In addIntersection from_node does not exist on edge");
+}
+
+void
+EFAEdge::resetIntersection(double position, EFANode * embedded_node_tmp, EFANode * from_node)
+{
+  for (unsigned int i = 0; i < _embedded_nodes.size(); ++i)
+  {
+    if (_embedded_nodes[i] == embedded_node_tmp)
+    {
+      if (from_node == _edge_node1)
+        _intersection_x[i] = position;
+      else if (from_node == _edge_node2)
+        _intersection_x[i] = 1.0 - position;
+      else
+        EFAError("In resetIntersection from_node does not exist on edge");
+      break;
+    }
+  }
+}
+
+void
+EFAEdge::copyIntersection(const EFAEdge & other, unsigned int from_node_id)
+{
+  _embedded_nodes.clear();
+  _intersection_x.clear();
+  _embedded_nodes = other._embedded_nodes;
+  if (from_node_id == 0)
+    _intersection_x = other._intersection_x;
+  else if (from_node_id == 1)
+  {
+    for (unsigned int i = 0; i < other.numEmbeddedNodes(); ++i)
+      _intersection_x.push_back(1.0 - other._intersection_x[i]);
+  }
+  else
+    EFAError("from_node_id out of bounds");
+  if (_embedded_nodes.size() != _intersection_x.size())
+    EFAError("in copyIntersection num emb_nodes must == num of inters_x");
+}
+
+EFANode *
+EFAEdge::getNode(unsigned int index) const
+{
+  if (index == 0)
+    return _edge_node1;
+  else if (index == 1)
+    return _edge_node2;
+  else
+    EFAError("In getNode index out of bounds");
+}
+
+void
+EFAEdge::reverseNodes()
+{
+  EFANode* tmp = _edge_node1;
+  _edge_node1 = _edge_node2;
+  _edge_node2 = tmp;
+  for (unsigned int i = 0; i < _embedded_nodes.size(); ++i)
+    _intersection_x[i] = 1.0 - _intersection_x[i];
+}
+
+bool
+EFAEdge::hasIntersection() const
+{
+  return (_embedded_nodes.size() > 0);
```

Comment:
The Code Standards don't say anything about that.  Is it a problem to have them?